llvm/tools/llvm-mca/llvm-mca.cpp

```diff
@@ -597,14 +610,25 @@ int main(int argc, char **argv) {
           // Default case.
           WithColor::error() << toString(std::move(NewE));
         }
+        if (SkipUnsupportedInstructions) {
+          DroppedInsts.insert(&MCI);
+          continue;
+        }
         return 1;
       }
 
       IPP->postProcessInstruction(Inst.get(), MCI);
-
+      InstToInstruments.insert({&MCI, Instruments});
```

Comment:
It isn't clear to me why this needs to be relocated. Could you please clarify?

Reply:
The `continue` above is dropping instructions, so it seemed logical to me to avoid inserting these dropped instructions into other data structures since they're now "gone".